### PR TITLE
Allow an admin to disable user changing email and username

### PR DESCRIPTION
**Pull request by ChrisMacNaughton**

### CODE REVIEW

File: app/controllers/admin/settings_controller.rb [hunk 1]
```diff
@@ -8,13 +8,18 @@ def index; end
...
   # PATCH/PUT /admin/settings/1.json
-  def update # rubocop:disable Metrics/MethodLength
+  def update # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
     opts = setting_params
     opts[:registration_enabled] = if opts[:registration_enabled].nil?
                                     false
                                   else
                                     true
                                   end
+    opts[:permemant_username] = if opts[:permemant_username].nil?
```
Comment (igalic) [inline app/controllers/admin/settings_controller.rb:18]: so `if opts[:permanent_username]` is `false`, we make it true?

File: app/views/devise/registrations/edit.html.erb [hunk 2]
```diff
@@ -2,9 +2,15 @@
 
 <%= settings_row("Account", "Your registered account details") do %>
     <%= bootstrap_form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
-      <%= f.text_field :username %>
-      <%= f.email_field :email, autofocus: true %>
-      <%= f.text_field :name %>
+        <%- if Setting.permemant_username %>
+          <%= f.text_field :username, disabled: true %>
```
Comment (igalic) [inline app/views/devise/registrations/edit.html.erb:6]: what happens when somebody edits the `disabled` thing in their browser?

**Review by igalic — COMMENTED** [review]
🙋🏻️

**Comment by ChrisMacNaughton** [comment]
The review comments were discussed independently

**Pull request merged**